python/python/glide/async_commands/core.py

```diff
@@ -2306,6 +2267,72 @@ async def geopos(
             await self._execute_command(RequestType.GeoPos, [key] + members),
         )
 
+    async def geosearch(
+        self,
+        key: str,
+        search_from: Union[str, GeospatialData],
+        seach_by: Union[GeoSearchByRadius, GeoSearchByBox],
+        order_by: Optional[OrderBy] = None,
+        count: Optional[GeoSearchCount] = None,
+        with_coord: bool = False,
+        with_dist: bool = False,
+        with_hash: bool = False,
+    ) -> List[Union[str, List[Union[str, float, int, List[float]]]]]:
+        """
+        Searches for members in a sorted set stored at `key` representing geospatial data within a circular or rectengal area.
+
+        See https://valkey.io/commands/geosearch/ for more details.
+
+        Args:
+            key (str): The key of the sorted set representing geospatial data.
+            search_from (Union[str, GeospatialData]): The location to search from. Can be specified either as a member
+                from the sorted set or as a geospatial data (see `GeospatialData`).
+            search_by (Union[GeoSearchByRadius, GeoSearchByBox]): The search criteria.
+                For circular area search, see `GeoSearchByRadius`.
+                For rectengal area search, see `GeoSearchByBox`.
+            order_by (Optional[OrderBy]): Specifies the order in which the results should be returned.
+                Defaults to None. See `OrderBy`.
+            count (Optional[GeoSearchCount]): Specifies the maximum number of results to return.
+                Defaults to None. See `GeoSearchCount`.
+            with_coord (bool): Whether to include coordinates of the returned items. Defaults to False.
+            with_dist (bool): Whether to include distance from the center in the returned items.
+                The distance is returned in the same unit as specified for the `search_by` arguments. Defaults to False.
+            with_hash (bool): Whether to include geohash of the returned items. Defaults to False.
+
+        Returns:
+            List[Union[str, List[Union[str, float, int, List[float]]]]]: If no with_* was set to True, returns a list of memebers
+            names.
+            If `with_coord`, `with_dist` or `with_hash` was specified, returns an array of arrays, we're each sub array represents a single item with:
```

Comment:
its a bit useless, since we would like to keep them all as a boolean and not some long enum, so user can still pass False to all of this parameters, so I think splitting is useless

Reply:
and always returning a 2 dimensional arrays seems not right